compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -3116,12 +3116,31 @@ fn add_lld_args(
 
     let self_contained_linker = self_contained_cli || self_contained_target;
     if self_contained_linker && !sess.opts.cg.link_self_contained.is_linker_disabled() {
-        for path in sess.get_tools_search_paths(false) {
-            cmd.arg({
-                let mut arg = OsString::from("-B");
-                arg.push(path.join("gcc-ld"));
-                arg
-            });
+        // We tell `cc` where to find the lld wrappers, in a similar way to how we locate
+        // codegen-backends dylibs:
+        // - in the tool search paths: the sysroot's `rustlib` bin path,
+        // - and as a fallback for cases where `--sysroot` only contains a target std: in the
+        //   default host sysroot where rustc is currently located.
+        let fallback_sysroot_paths = filesearch::sysroot_candidates()
+            .into_iter()
+            .map(|sysroot| filesearch::make_target_bin_path(&sysroot, config::host_triple()));
```

Comment:
Maybe this logic should go directly into `get_tools_search_paths`?

Reply:
I debated that as well, but didn't know if we wanted such a fallback in all cases, but it could be useful for other host tools and `--sysroot`. 

I'll move it there.

Reply:
Done. The fallback search paths will also added to the msvc paths, I guess that's fine. 

@rustbot ready